common/src/main/java/io/druid/metadata/MetadataStorageConnectorConfig.java

```diff
@@ -86,4 +86,47 @@ public String toString()
            ", passwordProvider=" + passwordProvider +
            '}';
   }
+
+  @Override
+  public boolean equals(Object o)
+  {
+    if (this == o) {
+      return true;
+    }
+    if (!(o instanceof MetadataStorageConnectorConfig)) {
+      return false;
+    }
+
+    MetadataStorageConnectorConfig that = (MetadataStorageConnectorConfig) o;
+
+    if (isCreateTables() != that.isCreateTables()) {
+      return false;
+    }
+    if (getPort() != that.getPort()) {
+      return false;
+    }
+    if (getHost() != null ? !getHost().equals(that.getHost()) : that.getHost() != null) {
+      return false;
+    }
+    if (getConnectURI() != null ? !getConnectURI().equals(that.getConnectURI()) : that.getConnectURI() != null) {
+      return false;
+    }
+    if (getUser() != null ? !getUser().equals(that.getUser()) : that.getUser() != null) {
+      return false;
+    }
+    return passwordProvider != null ? passwordProvider.equals(that.passwordProvider) : that.passwordProvider == null;
```

Comment:
What kind of problems might this cause if `PasswordProvider` does not properly implement `equals`?

Also making @himanshug aware of this